pkg/utils/kubernetes/ownerrefs.go

```diff
@@ -38,6 +38,13 @@ func SetOwnerForObject(obj, owner client.Object) {
 	}
 }
 
+// SetOwnerForObjectThroughLabel sets the owner of the provided object through a label.
+func SetOwnerForObjectThroughLabel(obj, owner client.Object, label string) {
+	labels := obj.GetLabels()
+	labels[label] = owner.GetName()
```

Comment:
Given that an owner might be a namespaced object (e.g. a ControlPlane) - aren't we losing here this information, putting only a name into the label? I think we also lose type information - we're not able to tell if an object labeled with `GatewayOperatorManagedByNameLabel` is a reference to a ControlPlane or some other type.

Reply:
👍 

We could save the type (by checking the kind of the owner) and ns as well. Moreover the `label` is always `consts.GatewayOperatorManagedByNameLabel` so I'd drop this as an argument here.

When more types need to be supported we can always use generics and type asserts (or stick to `GetObjectKind()` and set the labels accordingly.

Reply:
Yep, I've changed the code to use the following label set:

- `gateway-operator.konghq.com/managed-by`: `ControlPlane.gateway-operator.konghq.com`
- `gateway-operator.konghq.com/managed-by-name`: `<controlplane-name>`
- `gateway-operator.konghq.com/managed-by-namespace`: `<controlplane-namespace>`

Let me know what you think :)

Reply:
Regarding the `managed-by` label: currently we use this with just the uncapitalized kind name in the value, e.g. 

https://github.com/Kong/gateway-operator/blob/9015ff6fb46a2325bd5c63058d5c30bf58707e43/controller/controlplane/controller_reconciler_utils.go#L258

where

```
const (
	// ControlPlaneManagedLabelValue indicates that an object's lifecycle is managed
	// by the controlplane controller.
	ControlPlaneManagedLabelValue = "controlplane"
)
```

so the resource gets the following label:

```
gateway-operator.konghq.com/managed-by: gateway
```

I'd suggest that we stay consistent and keep it that way. WDYT?

Reply:
sure thing, I've made it consistent 👍